app/api/v1/endpoints/recommendations.py

```diff
@@ -0,0 +1,43 @@
+from flask import request
+from flask_restplus import Resource
+from app.api.v1.api import api
+from app.api.v1.models.recommendations import recommendation, recommendation_context, recommendation_parser
+from app.recommender_engine_stub import RecommenderEngineStub
+from app.user_activity_gatherer import UserActivityGatherer
+from app.recommender_engine_stub import InvalidRecommendationPanelLocationError, InvalidRecommendationPanelVersionError
+
+recommendation_name_space = api.namespace("recommendations", "Endpoint used for getting recommendations")
+
+
+@api.errorhandler(InvalidRecommendationPanelVersionError)
+def handle_root_exception(error):
+    return {'message': error.message()}, 400
+
+
+@api.errorhandler(InvalidRecommendationPanelLocationError)
+def handle_root_exception(error):
+    return {'message': error.message()}, 400
+
+
+@recommendation_name_space.route("")
+class Recommendation(Resource):
+    @api.expect(recommendation_context, recommendation_parser, validate=True)
+    @api.marshal_with(recommendation)
+    def post(self):
+        """Returns list of ids of recommended scientific services"""
+        args = recommendation_parser.parse_args()
+        location = args.get("location", None)
+        version = args.get("version", None)
+        context = request.get_json()
+
+        full_context = {
+            "user_id": context["user_id"],
+            "timestamp": "time",
+            "filters": context.get("filters", None),
+            "search_phrase": context.get("search_phrase", None)
```

Comment:
`None` is the default in `dict.get` so this here is a little to verbose. 
But it's not a mistake, extra verbosity can be useful.

Reply:
Fixed

Reply:
fixed